add-on/src/lib/redirect-handler/blockOrObserve.ts

```diff
@@ -320,10 +320,13 @@ export function addRuleToDynamicRuleSetGenerator (
     const state = getState()
     const redirectIsOrigin = originUrl === redirectUrl
     const redirectIsLocal = isLocalHost(originUrl) && isLocalHost(redirectUrl)
-    const badOriginRedirect = originUrl.includes(state.gwURL.host) && !redirectUrl.includes('recovery')
+    const localOriginRedirect = isLocalHost(originUrl) && !isLocalHost(redirectUrl)
+    // if the redirect is to the gateway and the automatic mode is off, we would like to present the recovery page.
+    // in case automatic mode is on, we would like to redirect to the public gateway.
```

Comment:
I don't think the status of automatic mode should impact/bypass the recovery page – we don't want to blindly redirect users to third-party server – see https://github.com/ipfs/ipfs-companion/issues/1204#issuecomment-1599105933